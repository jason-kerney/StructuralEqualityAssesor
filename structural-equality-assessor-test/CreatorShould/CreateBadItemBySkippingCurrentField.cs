using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using StructuralEqualityAssessor.Internal;
using StructuralEqualityAssessor.Test.Examples;
using StructuralEqualityAssessor.Test.Examples.LookUpFailingClasses;
using StructuralEqualityAssessor.Test.Examples.LookUpGoodClasses;
using Xunit;

namespace StructuralEqualityAssessor.Test.CreatorShould
{
    [UseReporter(typeof(DiffReporter))]
    public class CreateBadItemBySkippingCurrentField
    {
        [Fact]
        public void OfType()
        {
            Func<object> ctor = () => new AnotherClass();
            var (propertyValues, fieldValues) = Creator.GetPropertyAndFieldValues(typeof(AnotherClass), null);
            var result = Creator.CreateBadItemBasedOnFields(ctor, propertyValues, fieldValues, fieldValues.First().Key);

            Approvals.VerifyAll(result, "result");
        }

        [Fact]
        public void OfTypeDiffField()
        {
            Func<object> ctor = () => new AnotherClass();
            var (propertyValues, fieldValues) = Creator.GetPropertyAndFieldValues(typeof(AnotherClass), null);
            var result = Creator.CreateBadItemBasedOnFields(ctor, propertyValues, fieldValues, fieldValues.Last().Key);

            Approvals.VerifyAll(result, "result");
        }

        [Fact]
        public void OfTypeWithArrays()
        {
            Func<object> ctor = () => new HasArrays();
            var ctors = new Dictionary<Type, Func<object>>
            {
                {typeof(NonDefaultConstructor), () => new NonDefaultConstructor(77)},
            };

            var (propertyValues, fieldValues) = Creator.GetPropertyAndFieldValues(typeof(HasArrays), ctors);
            var result = Creator.CreateBadItemBasedOnFields(ctor, propertyValues, fieldValues, fieldValues.Last().Key);

            Approvals.VerifyAll(result, "result");
        }
    }
}