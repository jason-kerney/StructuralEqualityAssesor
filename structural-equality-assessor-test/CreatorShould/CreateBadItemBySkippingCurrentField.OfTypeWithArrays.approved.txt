result[0] = { Numbers: [1], Classes: [{ Thing: { SomeNumber: 1 }, Thing2: { ANumber: 1 }, aString: "A string", aChar: 'A' }], otherNumbers: [] }
result[1] = { Numbers: [1], Classes: [{ Thing: { SomeNumber: 1 }, Thing2: { ANumber: 1 }, aString: "A string", aChar: 'A' }], otherNumbers: null }
